### PR TITLE
Updated Java to version 17 in CI

### PR DESCRIPTION
**Pull request by odisseus**
This is a prerequisite for https://github.com/sourcegraph/cody-issues/issues/240.

## Test plan

Green CI

### CODE REVIEW

**Review by pkukielka — APPROVED** [review]
LGTM

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -10,10 +10,10 @@ jobs:
       - uses: actions/setup-node@v3
         with:
           node-version: 20
-      - uses: actions/setup-java@v3
+      - uses: actions/setup-java@v4
         with:
           distribution: "temurin"
-          java-version: 11
```
Comment (mkondratek) [inline .github/workflows/release.yml:16]: so the plugin will be built with java 17 and published. will a user be able to install it in IntelliJ 2022.1?

Comment (odisseus) [reply .github/workflows/release.yml:16]: For compatibility with 2022.1, we need to produce Java 11 bytecode. However, this is [configured](https://github.com/sourcegraph/jetbrains/blob/main/gradle.properties#L18) in a different place.

Comment (mkondratek) [reply .github/workflows/release.yml:16]: ack, thanks

**Pull request merged**